Bugfixes

DrDudelsack opened the pull request.

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Create test

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Create test

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Translations by PKZB

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Characters.java:
```diff
@@ -1,210 +1,210 @@
-package ImperatorToCK2; 
```
Leave the package definition in.

DrDudelsack replied:
That might be why it's not building, lol, I think that my IDE removed them. I'll add them back tomorrow

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Directories.java:
```diff
@@ -1,129 +1,132 @@
-package ImperatorToCK2;
```
Leave the package definition in.

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Importer.java:
```diff
@@ -1,948 +1,981 @@
-package ImperatorToCK2;
```
Leave the package definition in.

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Main.java:
```diff
@@ -1,689 +1,754 @@
-package ImperatorToCK2;
```
Leave the package definition in.

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Output.java:
```diff
@@ -1,752 +1,805 @@
-package ImperatorToCK2;
```
Leave the package definition in.

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Processing.java:
```diff
@@ -1,737 +1,749 @@
-package ImperatorToCK2;
```
Leave the package definition in.

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Output.java:
```diff
@@ -1,752 +1,805 @@
+        return ck2CultureInfo;
+    }
+    
+    public static void logPrint(String name) throws IOException //outputs to log.txt
```
Unfortunately this will clash with the logger stuff I added. :-(

I'd suggest going with what I added for two reasons:
1 - It uses a standard java logging library, and standard is preferred.
2 - It matches the format the fronter expects, whereas this would need some more work to do so.

It's good work though. Sorry for doing something at the same time without checking!

Idhrendur replied:
You can replace calls to logPrint with one of:
```
LOGGER.severe() for things that'll break the converter run entirely (or just throw an uncaught exception for it to be logged automatically)
LOGGER.warning() for bad things that you want users to be aware of, but that don't prevent the conversion from finishing
LOGGER.info() for useful information to have in the log (such as switch from one phase of conversion to another)
LOGGER.config() for debug messages. Not useful now, but handy in the future if you add a debug mode option.
LOGGER.finest() to move the status bar in the fronter. Give it things like "25%" or "75%"
```

DrDudelsack replied:
That sounds good, I'll replace those logPrints with LOGGER sometime tomorrow

Idhrendur commented on ImperatorToCk2/src/main/java/ImperatorToCK2/Main.java:
```diff
@@ -1,689 +1,754 @@
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
-import ImperatorToCK2.ConverterLogger;
```
You probably need this import.

DrDudelsack replied:
`import ImperatorToCK2.ConverterLogger` actually causes an error, as long as the .java file is in there with the other .java files, import statements don't need to be added

DrDudelsack committed:
Logger, bugfixes, misc tweaks

Readded package ImperatorToCK2, replaced all instances of logPrint() with LOGGER, more logger messages, log messages organized into appropriate levels, made Hindu, Buddhist, and Jain characters have Kshatriya caste trait, and fixed typo with 100 AD start date spouses
